CONTRIBUTING.md

```diff
@@ -23,12 +23,101 @@ tox run -e integration   # integration tests
 tox                      # runs 'format', 'lint', 'static', and 'unit' environments
 ```
 
+## Recommended development environment setup
+
+The recommended development environment is set up on `multipass`.
+
+Get it via:
+
+```shell
+sudo snap install multipass --classic
+```
+
+Steps to setting up the environment:
+```shell
+multipass launch -c 4 -m 16G -d 50G -n charm-dev charm-dev
+multipass shell charm-dev
+# Refer to [1] below for a note.
+juju switch lxd
+juju add-model datahub-vm lxd
+juju switch microk8s
+juju add-model datahub-k8s microk8s
+# Refer to [2] below for a note.
+```
+
+[1]: Minor and/or patch versions of the tech stack can introduce regressions, if you are having problems use the following commands to switch to tested channels:
+```shell
+sudo snap switch juju --channel 3.4/stable
+sudo snap refresh juju
+sudo snap switch microk8s --channel 1.29/stable
+sudo snap refresh microk8s --classic
+```
+
+Afterwards, you will have to recreate the controllers and models from scratch:
+```shell
+juju bootstrap lxd lxd
+juju add-model datahub-vm
+juju bootstrap microk8s microk8s
+juju add-model datahub-k8s microk8s
+```
+
+[2]: DataHub has Opensearch as a dependency. In order to deploy the Opensearch charm, you need to edit certain kernel parameters inside your `multipass` instance as instructed [here](https://charmhub.io/opensearch/docs/t-set-up#heading--kernel-parameters).
+
+## Deploy the dependencies
+
+DataHub has PostgreSQL, Kafka, and OpenSearch as dependencies. Opensearch only has a machine charm, while the others have both a machine and a Kubernetes charm. In production, all of the dependencies are intended to be deployed on machines but for simplification of development purposes you can have only Opensearch on a machine and deploy the rest on `microk8s`. So, the instructions here will cover that scenario.
+
+Deploying Opensearch:
+```shell
+juju switch lxd:datahub-vm
+juju deploy opensearch --channel 2/edge -n 3
```

Comment:
Do we need 3 units of Opensearch for a dev setup?

Reply:
The charm does not start with a single unit and I went for an odd number of units.